azure/callback.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Callback, CallbackFactory, Context, EventSubscription, EventSubscriptionArgs } from "@pulumi/azure-serverless/subscription";
+import * as cloud from "@pulumi/cloud";
+import * as pulumi from "@pulumi/pulumi";
+import * as shared from "./shared";
+
+/**
+ * Azure-specific data to create an AWS lambda out of a callback function.  Can be passed to any
+ * functions in pulumi/cloud-aws that can take a cloud.Callback<T> argument.  To create the same
+ * default AwsCallbackData Pulumi creates when given a simple JavaScript function, use
+ * [createCallbackData].
+ */
+export interface AzureCallbackData<T extends Function> extends cloud.CallbackData<T>, EventSubscriptionArgs<any, any> {
+    /**
+     * Not used.  Provide [function] instead.
+     */
+    func?: never;
+    /**
+     * Not used.  Provide [function] instead.
+     */
+    factoryFunc?: never;
+}
```

Comment:
same concepts apply to azure.  this also deviates from aws currently in that EventSubscriptionArgs is at azure-serverless, not pulumi/azure.  That will be rectified this sprint.